wperf/arg-parser-arg.cpp

```diff
@@ -0,0 +1,219 @@
+    {
+        return m_description;
+    }
+
+    arg_parser_arg arg_parser_arg::add_alias(std::wstring new_alias)
+    {
+        m_aliases.push_back(new_alias);
+        return *this;
+    }
+
+    size_t arg_parser_arg::get_arg_count() const
+    {
+        return m_arg_count;
+    }
+
+    std::wstring arg_parser_arg::get_name() const
+    {
+        return m_name;
+    }
+
+    std::wstring arg_parser_arg::get_alias_string() const
+    {
+        // convert alias vector to wstring
+        std::wstring alias_string;
+        for (auto& m_alias : m_aliases) {
+            if (m_alias.empty()) continue;
+            alias_string.append(m_alias + L", ");
+        }
+        if (alias_string.find(L", ") != std::wstring::npos) alias_string.erase(alias_string.end() - 2, alias_string.end());
+
+        return alias_string;
+    }
+
+    arg_parser_arg arg_parser_arg::add_check_func(std::function<bool(const std::wstring&)> check_func)
+    {
+        m_check_funcs.push_back(check_func);
+        return *this;
+    }
+
+    void arg_parser_arg::set_is_parsed()
+    {
+        m_is_parsed = true;
+    }
+
+    bool arg_parser_arg::is_parsed() const
+    {
+        return m_is_parsed;
+    }
+
+    bool arg_parser_arg::is_set() const
+    {
+        return is_parsed() && m_values.size() == m_arg_count;
+    }
+
+    std::vector<std::wstring> arg_parser_arg::get_values() const
+    {
+        return m_values;
+    }
+
+    bool arg_parser_arg::parse(std::vector<std::wstring> arg_vect)
+    {
+        if (arg_vect.size() == 0 || !is_match(arg_vect[0]))
+            return false;
+
+        if (m_arg_count == -1 && arg_vect.size() > 0) m_arg_count = arg_vect.size() - 1;
+
+        if (arg_vect.size() < m_arg_count + 1)
+            throw std::invalid_argument("Not enough arguments provided.");
+
+        if (m_arg_count == 0)
+        {
+            set_is_parsed();
+            return true;
+        }
+
+        for (int i = 1; i < m_arg_count + 1; ++i)
+        {
+            for (auto& check_func : m_check_funcs)
+            {
+                if (!check_func(arg_vect[i]))
+                    throw std::invalid_argument("Invalid arguments provided.");
+            }
+            m_values.push_back(arg_vect[i]);
+        }
+        set_is_parsed();
+        return true;
+    }
+
+
+    std::wstring arg_parser_add_wstring_behind_multiline_text(const std::wstring& str, const std::wstring& prefix)
+    {
+        std::wstring formatted_str;
+        std::wstring current_line;
+        std::wistringstream iss(str);
+        while (getline(iss, current_line, L'\n'))
+        {
+            if (current_line.empty())
+            {
+                formatted_str += L"\n";
+                continue;
+            }
+
+            formatted_str += prefix + current_line + L"\n";
+        }
+        return formatted_str;
+    }
+
+    std::wstring arg_parser_format_string_to_length(const std::wstring& str, size_t max_width)
```

Comment:
```cpp
// At least 1 line of text what this function should do.
```

Reply:
That is represented in the `arg-parser-arg-utils-tests` file